### ISSUE TITLE
[require-await] False positive on async generator that uses yield*

### ISSUE THREAD

**Issue by felixfbecker** [typescript-eslint/typescript-eslint#1691]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/require-await": "error"
  }
}
```

```TS
async function* foo() {
  await Promise.resolve()
  yield 1
}
async function* bar() {
  yield* foo()
}
```

**Expected Result**
No error. `bar` has to be async because it delegates to `foo`, which is async.

**Actual Result**
Error: Async generator function has no 'await' expression.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |
